### ISSUE TITLE
Style/RedundantCondition autofixes incorrectly (rubocop 1.29.1)

### ISSUE THREAD

**Issue by delphaber** [rubocop/rubocop#10636]
This code

```ruby
class Pippo
  def foo
    result =
      if x
        X.find(x)
      else
        Y.find(y)
      end

    result.to_s
  end
end
```

generates this warning:

```bash
app/pippo.rb:4:7: C: [Correctable] Style/RedundantCondition: Use double pipes || instead.
      if x ...
```

If I run `rubocop -a` it gets autofixed with:

```ruby
class Pippo
  def foo
    result =
      X.find(x || y)

    result.to_s
  end
end
```

but the behavior now it's changed.

## Expected behavior

I think the cop should not raise a warning, because it's not a redundant condition 🤔 

## RuboCop version

```
$ bundle exec rubocop -V
 ↳ $ bundle exec rubocop -V
1.29.1 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 2.7.6 x86_64-darwin21)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.10.0
```

**Comment by delphaber** [comment]
thank you! <3
